lib/src/DOtherSide.cpp

```diff
@@ -135,12 +137,53 @@ ::DosQQmlContext *dos_qqmlapplicationengine_context(::DosQQmlApplicationEngine *
     return engine->rootContext();
 }
 
+void dos_qqmlapplicationengine_addimageprovider(DosQQmlApplicationEngine *vptr, DosQQuickImageProvider *vptr_i)
+{
+  auto engine = static_cast<QQmlApplicationEngine *>(vptr);
+  auto provider = static_cast<QQuickImageProvider *>(vptr_i);
+  engine->addImageProvider(QString("test"), provider);
+}
+
 void dos_qqmlapplicationengine_delete(::DosQQmlApplicationEngine *vptr)
 {
     auto engine = static_cast<QQmlApplicationEngine *>(vptr);
     delete engine;
 }
 
+
+::DosQQuickImageProvider *dos_qquickimageprovider_create()
+{
+    return new DosImageProvider();
```

Comment:
When casting to void* and back you should always use the same class. So probably the right way shoud be to write

return static_cast<QQuickImageProvider*>(new DosImageProvider())

Reply:
In other words you must be consistent:
if you cast from A* to void* than you should cast from void* to A*